packages/material-ui/src/ExpansionPanel/ExpansionPanel.test.js

```diff
@@ -115,6 +115,22 @@ describe('<ExpansionPanel />', () => {
     assert.strictEqual(wrapper.hasClass(classes.disabled), true);
   });
 
+  it('should handle the CollapseComponent prop', () => {
+    const CustomCollapse = props => <div {...props} />;
+
+    const wrapper = shallow(
+      <ExpansionPanel CollapseComponent={CustomCollapse}>
+        <ExpansionPanelSummary />
+        <div>Hello</div>
+      </ExpansionPanel>,
+    );
+
+    assert.strictEqual(wrapper.childAt(0).type(), ExpansionPanelSummary);
```

Comment:
We're trying to move away from these kind of tests. They're extremely brittle. Overall your test is testing  implementation rather than what the user interacts with.

Could you add a test that is closer to a real-world use case? Doesn't have to be a complete integration but at least something different than "at this exact location has to be this component".

Reply:
Hopefully https://github.com/mui-org/material-ui/pull/14617/commits/9e1937063b00f56ced8b370f3a48adf445eba1b4 does the trick ;)